src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -240,23 +261,27 @@
                                                                             })
                                                                             .collect(Collectors.toList());
 
-        // Create all VMs in parallel and collect IDs
+        // Execute init scripts asynchronously on VMs
+        //vms.forEach(vm -> executeScriptOnVM(vm, instance.getInitScript()));
+
+        // Create all VMs in parallel
         return azureService.virtualMachines()
                            .create(creatableVirtualMachines)
                            .values()
                            .stream()
+                           // Use vmId() instead of id() for clarity (id() contains full resource path)
                            .map(vm -> instance.withTag(vm.name()).withId(vm.vmId()).withNumber(SINGLE_INSTANCE_NUMBER))
                            .collect(Collectors.toSet());
     }
 
     protected Creatable<NetworkInterface> createPublicAddressAndNetworkInterface(Azure azureService, String instanceTag,
             ResourceGroup resourceGroup, Region region, AzureNetworkOptions networkOptions, int instanceNumber) {
         // Create a new public IP address (one per VM)
-        String publicIPAddressName = createUniquePublicIPName(createUniqueInstanceTag(instanceTag, instanceNumber));
-        Creatable<PublicIpAddress> creatablePublicIpAddress = azureProviderNetworkingUtils.preparePublicIPAddress(azureService,
+        String PublicIPAddressName = createUniquePublicIPName(createUniqueInstanceTag(instanceTag, instanceNumber));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)